static/js/components/dashboard/CourseListCard.js

```diff
@@ -11,54 +11,179 @@ import type {
 import FinancialAidCalculator from '../../containers/FinancialAidCalculator';
+import type { CoursePrice } from '../../flow/dashboardTypes';
+import type { CourseRun } from '../../flow/programTypes';
+import {
+  FA_TERMINAL_STATUSES,
+  COUPON_CONTENT_TYPE_PROGRAM,
+  COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT,
+  COUPON_AMOUNT_TYPE_FIXED_DISCOUNT,
+} from '../../constants';
+import {
+  makeAmountMessage,
+  makeCouponReason,
+  isFreeCoupon,
+} from '../../lib/coupon';
+import { pickExistingProps } from '../../util/util';
+
+const priceMessageClassName = "price-message";
 
 export default class CourseListCard extends React.Component {
   props: {
-    coupon:                       ?Coupon,
+    coupon?:                      Coupon,
     program:                      Program,
-    courseEnrollAddStatus?:       string,
+    coursePrice:                  CoursePrice,
     prices:                       CalculatedPrices,
     openFinancialAidCalculator?:  () => void,
     now?:                         Object,
-    addCourseEnrollment:          (courseId: string) => void,
-    openCourseContactDialog:      (course: Course, canContactCourseTeam: boolean) => void
+    addCourseEnrollment:          (courseId: string) => Promise<*>,
+    openCourseContactDialog:      (course: Course, canContactCourseTeam: boolean) => void,
+    setEnrollSelectedCourseRun:   (r: CourseRun) => void,
+    setEnrollCourseDialogVisibility: (bool: boolean) => void,
   };
 
+  renderFinancialAidPriceMessage() {
+    const {
+      program: {financial_aid_user_info: financialAid},
+      coupon,
+      prices,
+    } = this.props;
+
+    if (FA_TERMINAL_STATUSES.includes(financialAid.application_status)) {
+      // all prices in the map are the same, so just get a random one
+      const iter = prices.values();
+      const { value: price } = iter.next();
+      if (!price) {
+        return null;
+      }
+      if (coupon && coupon.content_type === COUPON_CONTENT_TYPE_PROGRAM) {
+        // financial aid + coupon
+        return <p className={priceMessageClassName}>
+          Your price is <strong>${ price.toString() } USD per course,</strong> including
+          both financial aid and your coupon.
+          If you want to audit courses for FREE and upgrade later,
+          click Enroll Now then choose the audit option.
+        </p>;
+      } else {
+        return <p className={priceMessageClassName}>
+          Your Personal Course Price is{" "}
+          <strong>${ price.toString() } USD per course.</strong> {" "}
+          If you want to audit courses for FREE and upgrade later, click
+          Enroll Now then choose the audit option.
+        </p>;
+      }
+    } else {
+      return <p className={priceMessageClassName}>
+        You need to get your Personal Course Price before you can pay for courses.
+        If you want to audit courses for FREE and upgrade later,
+        click Enroll Now then choose the audit option.
+      </p>;
+    }
+  }
```

Comment:
I think this should have an explicit `return null;` at the end for FA applications not in the states above.